docs/api/index.rst

```diff
@@ -0,0 +1,120 @@
+.. SPDX-License-Identifier: MIT
+
```

Comment:
IMO, all of the subsections in this file should be ordered EITHER alphabetically OR the same way that they ordered internally in each file. @shiftinv which way do you think they should be ordered?

Reply:
Matching the order in the other files would be better, assuming that's consistent everywhere. Could perhaps even move these definitions to a separate page, and only keep the documents list?

Reply:
[`6b5a509`](https://github.com/DisnakeDev/disnake/pull/722/commits/6b5a509ed427d38fca66f07ea10bbd003648cd62)